subscription.go

```diff
@@ -103,6 +103,12 @@ func (r *marathonClient) registerSubscription() error {
 	case EventsTransportCallback:
 		return r.registerCallbackSubscription()
 	case EventsTransportSSE:
+		if timeout := r.client.config.HTTPClient.Timeout; timeout > 0 {
```

Comment:
No, user will get error only if he configures own http client with Timeout set and starts SSE subscription. I dont think silencing timeout is good approach in this situation as it will work in most cases (as default go client) but will add undefined behaviour when connection to marathon starts to misbehave (manifested by flaky SSE test under windows) - failing fast will enable user to set appropriate timeout w.r.t. to sse implementation or create 2 clients for different purposes.